### PR TITLE
Remove deprecated flag --resource-container from kube-proxy

### LINKED ISSUE #78287: Remove deprecated kube-proxy flag —resource-container
The —resource-container flag has been deprecated for many releases and can be removed. (TODO add full context)

### PR DESCRIPTION
**Pull request by vllry**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Removes a long-deprecated field (`--resource-container`) from kube-proxy. This also removes the resourceContainer code, as kube-proxy was the last component to use it.

**Which issue(s) this PR fixes**:
Fixes #78287

**Special notes for your reviewer**:
* If the user keeps the flag (EG as part of their bootstrap process), kube-proxy will fail to start, but the result may not be obvious. Is it okay to yank it all at once, or should we give a stronger warning when the flag is used and wait another release?
* Should we worry about the API change compatibility? As far as I can tell, removing a deprecated admin flag/v1alpha API field is fine.

**Does this PR introduce a user-facing change?**:

```release-note
ACTION REQUIRED: Removed deprecated flag `--resource-container` from kube-proxy.

The deprecated `--resource-container` flag has been removed from kube-proxy, and specifying it will now cause an error.  The behavior is now as if you specified `--resource-container=""`.  If you previously specified a non-empty `--resource-container`, you can no longer do so as of kubernetes 1.16.
```

### CODE REVIEW

**Comment by vllry** [comment]
/unassign @bowei 

Sorry for the spam :C

**Comment by vllry** [comment]
/uncc @bowei

Sorry for the spam :C (Of course I get this command wrong several times...)

File: pkg/proxy/apis/config/types.go [hunk 1]
```diff
@@ -138,9 +138,6 @@ type KubeProxyConfiguration struct {
 	// portRange is the range of host ports (beginPort-endPort, inclusive) that may be consumed
 	// in order to proxy service traffic. If unspecified (0-0) then ports will be randomly chosen.
 	PortRange string
-	// resourceContainer is the absolute name of the resource-only container to create and run
-	// the Kube-proxy in (Default: /kube-proxy).
-	ResourceContainer string
```
Comment (johscheuer) [inline pkg/proxy/apis/config/types.go]: You must also remove the default value for `ResourceContainer` (https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/apis/config/v1alpha1/defaults.go#L61-L63)

**Comment by vllry** [comment]
/assign @thockin 

I see you're still on Github, want to go 3/3 on ripping out flags? :D

File: pkg/proxy/apis/config/types.go (hunk 1 above)
Comment (vllry) [reply pkg/proxy/apis/config/types.go]: Thanks, I spotted that but I'm still missing something. All unit tests are fine, but the system doesn't start properly in e2e.

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

**Comment by nikhita** [comment]
FYI added ` ```release-note ` in the release note block to fix the release-note label.

**Comment by justinsb** [comment]
Can we change the release note flag to be more instructive?

I suggest:

```
ACTION REQUIRED: Removed deprecated flag `--resource-container` from kube-proxy.

The --resource-container flag has been removed from kube-proxy.  The behaviour is now as if you specified `--resource-container=""`.  However, you must stop specifying `--resource-container=""`.  If you previously specified a non-empty `--resource-container`, you can no longer do so as of kubernetes 1.16.
```

**Comment by thockin** [comment]
Updated

**Comment by zouyee** [comment]
https://github.com/kubernetes/kubernetes/issues/66614

**Pull request merged**